player/stream-test/js/script.js

```diff
@@ -128,44 +239,324 @@ function check404(url, callbackFunction) {
+    log("On Audio Playback Quality Changed: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_VIDEO_PLAYBACK_QUALITY_CHANGED, function (data) {
+    log("On Video Playback Quality Changed: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_STALL_STARTED, function (data) {
+    log("On Buffering Started: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_STALL_ENDED, function (data) {
+    log("On Buffering Ended: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_PLAYING, function (data) {
+    log("On Playing: " + JSON.stringify(data))
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_PAUSE, function (data) {
+    log("On Pause: " + JSON.stringify(data))
+  });
+
+  player.addEventHandler(player.EVENT.ON_PAUSED, function (data) {
+    log("On Paused: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_PLAY, function (data) {
+    log("On Play: " + JSON.stringify(data))
+  });
+
+  player.addEventHandler(player.EVENT.ON_METADATA_PARSED, function (data) {
+    log("On Metadata Parsed: " + JSON.stringify(data))
+  });
+
+  player.addEventHandler(player.EVENT.ON_READY, function (data) {
+    log("On Ready: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_SOURCE_LOADED, function (data) {
+    log("On Loaded: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_STOPPED, function (data) {
+    log("On Stopped: " + JSON.stringify(data));
+  });
+
+  player.addEventHandler(player.EVENT.ON_ERROR, function (data) {
+    log("On Error: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_SEEK, function (data) {
+    log("On Seek Started: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler(player.EVENT.ON_SEEKED, function (data) {
+    log("On Seek Finished: " + JSON.stringify(data));
+    updateCharts(player);
+  });
+
+  player.addEventHandler('onTimeChanged', function () {
+    updateCount++;
+
+    if (updateCount % 4 == 1) {
+      updateCharts(player);
+    }
+  });
+}
+
+
+function log(message) {
+  $('<p class="log-message"></p>').append(getTimestamp() + ' - ').append(message).appendTo('#logContent');
+}
+function checkTime(i) {
+  return (i < 10) ? "0" + i : i;
+}
+
+function getTimestamp() {
+  var now = new Date();
+  var h = checkTime(now.getHours());
+  var m = checkTime(now.getMinutes());
+  var s = checkTime(now.getSeconds());
+  var mm = checkTime(now.getMilliseconds());
+  now = h + ":" + m + ":" + s + ":" + mm;
+  return '<span class="timestamp">' + now + '</span>';
+}
+
+setupChart();
```

Comment:
setupChart only sets up the chart, but where are the other mechanism started (like continuous receiving of data)?

Reply:
You have `updateCharts` for that